### ISSUE TITLE
Hostname change on interface causes raspberry os lite to fail on boot

### ISSUE THREAD

**Issue by GloriousGloria23** [openmediavault/openmediavault#1741]
**Describe the bug**
Installed omv on Raspberry os lite.  After setup, I changed the hostname in the UI. System went into reboot but got stuck and did not finish. ssh failed to connect.
Even after directly connecting monitor, system was not responsive and screen turned black after initial boot sequence. 
Mounted the drive on another linux machine, manually reverting the hostname in the /etc/hostname file solved the issue and system rebooted normally.

**To Reproduce**

1. Go to omv -> Network -> General -> Hostname
2. Change Hostname
3. click Save
4. automated reboot failed

**Expected behavior**

successful reboot with new hostname

**openmediavault Server (please complete the following information):**

- OS version: Linux 6.6.20+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux
- openmediavault version:  7.0.5-1

**Client (please complete the following information):**

- Device: PC
- OS: Win 11
- Browser firefox
- Version 124

**Comment by votdev** [comment]
OMV is more or less exactly doing the same like raspi-config.

- https://github.com/RPi-Distro/raspi-config/blob/351366eaf28416fce1253a7a2569457e3d7570a4/raspi-config#L855
- https://github.com/openmediavault/openmediavault/blob/0a326a200927cce1c333ee561537882bee86fc0b/deb/openmediavault/srv/salt/omv/deploy/hostname/default.sls#L25
- https://github.com/openmediavault/openmediavault/blob/0a326a200927cce1c333ee561537882bee86fc0b/deb/openmediavault/srv/salt/omv/deploy/hosts/10default.sls#L34

If the problem is only on Raspi hardware, then it will be hard for me to reproduce and fix it without having such hardware. 
In that case the community must provide a fix.

**Comment by votdev** [comment]
Can not reproduce this issue. So it seems to me that this problem is specific to your system or Raspi OS based systems in special.

**Comment by votdev** [comment]
Where exactly does the system hang? What service is not starting or reports an error?

**Comment by GloriousGloria23** [comment]
I wish I knew, the problem is that as soon as it happened I completely lost access to the device. Ping would not respond (and therefore I couldn't SSH into the machine), and hooking it up to a monitor ended up with a blank screen (I tried two different monitor). I tried taking a video to see where it would hang but it went so fast it came out blurry.

Is there a way to find out where it hangs by reading a log file? I can try to reproduce it again on a new SD card sometimes next week (I dare not do it again on my running NAS :D )

**Comment by votdev** [comment]
You can use `journalctl -b` to get the last boot log or use `journalctl --since today` for the whole day.

**Comment by jochman** [comment]
I'm joining here. Same on RPI5 with PI OS Lite and OMV6.

**Comment by flmaxey** [comment]
Are you building your devices in accordance with -> [this process](https://wiki.omv-extras.org/doku.php?id=omv7:raspberry_pi_install)?

In the essentials, after installing an R-PI image with SSH enabled, the build is as follows:

**sudo apt-get update

sudo apt-get upgrade -y

wget -O - https://github.com/OpenMediaVault-Plugin-Developers/installScript/raw/master/preinstall | sudo bash

sudo reboot

wget -O - https://github.com/OpenMediaVault-Plugin-Developers/installScript/raw/master/install | sudo bash**

**Comment by rvegele** [comment]
Seems like I just ran into the same issue. 

Pi4, RPI lite (headless), with SSH. 

Can't connect (boot up?) after performing hostname change / save.

**Comment by ryecoaaron** [comment]
if you are using dhcp, it might have changed ip address.  Sometimes, you have to plugin a monitor in to see what is going on.

**Comment by rvegele** [comment]
Right, I see no pi in my router device list.

Here's where it stopped:
![Untitled](https://github.com/openmediavault/openmediavault/assets/8513898/0d53f3a5-de27-430b-ad9f-edc946612277)

And afterwards: 
![Untitled2](https://github.com/openmediavault/openmediavault/assets/8513898/853b3906-6dc0-4df1-887b-97af7807d195)

Not sure why address is 172... was 192.168... beforehand. So it's not communicating with the router somewhere any more as far as I can tell.

**Comment by ryecoaaron** [comment]
172. is the docker address not the system's address.  The login message isn't always up to date because dhcp can take too long sometimes.  Login and type: ip a

**Comment by rvegele** [comment]
hm... I'm connected through cable to my router :/ 

![IMG_7023](https://github.com/openmediavault/openmediavault/assets/8513898/bc682ad4-483d-4815-8412-d12a1d599450)

**Comment by ryecoaaron** [comment]
end0 is the network interface and it does not have an address.  I would run omv-firstaid

**Comment by rvegele** [comment]
running 1 - network config in ovm-firstaid produces 

![IMG_7024](https://github.com/openmediavault/openmediavault/assets/8513898/f4750c4d-9100-4ea0-98b1-4d6785f488cc)

**Comment by votdev** [comment]
You need to run it as root user.

# sudo omv-firstaid

**Comment by ryecoaaron** [comment]
It shouldn't break again unless you change the hostname again.  Some routers do dumb things when a mac address gets a new name.

**Comment by rvegele** [comment]
interestingly enough this has also knocked down my docker setup with jellyfin :(

**Comment by ryecoaaron** [comment]
You should post on the forum for help.  Many more people see posts there and I don't use jellyfin.

**Comment by rvegele** [comment]
> You should post on the forum for help. Many more people see posts there and I don't use jellyfin.

thank you for your help

**Comment by roblatour** [comment]
I recently had a similar problem, when I change the host name on the Pi running OMV I could no longer access it using ssh from my windows machine.  To correct, I issued the following command from the windows cmd window:

ssh-keygen -R x.x.x.x
(where x.x.x.x was the ip address of the Pi running OMV)

after that I could ssh into it again.
May be related?

**Comment by ryecoaaron** [comment]
ssh-keygen -R just removes that ip from the known_hosts file on the client.  This shouldn't prevent logging in via ssh.

**Comment by votdev** [comment]
> ssh-keygen -R just removes that ip from the known_hosts file on the client. This shouldn't prevent logging in via ssh.

The SSH `known_hosts` file does not explicitly contain the hostname alone; rather, it contains a hash of the hostname (or IP address) and the corresponding public key of the remote host. The purpose of this file is to prevent man-in-the-middle attacks by ensuring that the server you are connecting to is indeed the one you have connected to previously.

Here's how it works:

1. **Host Identification**: The `known_hosts` file typically includes the hostname or IP address (hashed or unhashed) and the server's public key. When you connect to a remote server, SSH checks this file to verify the server's identity.

2. **Hostname Change**: If the hostname changes, SSH will not find a matching entry in the `known_hosts` file because the hash of the new hostname does not match any existing entries. This mismatch will prompt a warning, and SSH will ask you to confirm the new host's identity, potentially adding the new hostname and its public key to the `known_hosts` file if you accept it.

3. **Server Key Change**: If the hostname remains the same but the server's public key changes (for example, if the server is rebuilt or its SSH keys are regenerated), SSH will detect a mismatch between the known public key for that hostname and the new key. This also results in a warning, and SSH will prevent the connection until you resolve the discrepancy, typically by updating the `known_hosts` file.

In summary, the `known_hosts` file helps ensure that you are connecting to the same server you connected to previously by matching both the hostname and the server's public key. If either the hostname or the server's public key changes, SSH will raise a warning and may prevent the connection until the change is acknowledged and accepted by the user.

**Comment by roblatour** [comment]
@votdev yes, that is right.  For more background, after changing the hostname on the Pi, when I tried to connect I got the error message "WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!"  followed by further explanation.  
I since done a little more searching and found this article ... https://forums.raspberrypi.com/viewtopic.php?t=362325 which offers more detail as as well as the same ssh-keygen -R x.x.x.x solution.
All to say, there may be a tie to changing the host name and other issues such as this one.
As I said in my original post "May be related (/may not be related)?"
Regardless, hope this is helpful.

**Comment by ryecoaaron** [comment]
> All to say, there may be a tie to changing the host name and other issues such as this one.

If you change the name of your OMV system, it can't change the client though.  So, you are forced to clean up the known_hosts file manually or with ssh-keygan.

**Comment by FPedrocas** [comment]
same thing just happened to me on a fresh install
Using Raspberry Pi OS Lite (64 bit) on a Pi 4 model B

**Comment by joaquinmartis** [comment]
I faced the same error after changing the hostname on the omv workbench.
The webpage turned black with a red sign telling me about an error in the system.
Restarting the RPi showed that a service related to networking was failing to start, thats why the RPi was not accesible through SSH.

Using sudo omv-firstaid and using the first option i was able to reconnect the RPi to my home network and to get it to work.

As someone said it is probable a problem within the router and not OMV or the RPi

**Comment by votdev** [comment]
The hostname is validated:

- https://github.com/openmediavault/openmediavault/blob/cb5010451521237239b33986291d6fc03cfbf088/deb/openmediavault/workbench/src/app/shared/forms/custom-validators.ts#L53
- https://github.com/openmediavault/openmediavault/blob/cb5010451521237239b33986291d6fc03cfbf088/deb/openmediavault/workbench/src/app/pages/network/general/general-network-form-page.component.ts#L45

The hostname spec can be found here: https://en.wikipedia.org/wiki/Hostname

**Comment by laleeroy** [comment]
> **Describe the bug** Installed omv on Raspberry os lite. After setup, I changed the hostname in the UI. System went into reboot but got stuck and did not finish. ssh failed to connect. Even after directly connecting monitor, system was not responsive and screen turned black after initial boot sequence. Mounted the drive on another linux machine, manually reverting the hostname in the /etc/hostname file solved the issue and system rebooted normally.
> 
> **To Reproduce**
> 
> Steps to reproduce the behavior:
> 
> 1. Go to omv -> Network -> General -> Hostname
> 2. Change Hostname
> 3. click Save
> 4. automated reboot failed
> 
> **Expected behavior**
> 
> successful reboot with new hostname
> 
> **openmediavault Server (please complete the following information):**
> 
> * OS version: Linux 6.6.20+rpt-rpi-v8 [Prevent fatal error in autoloader #1](https://github.com/openmediavault/openmediavault/pull/1) SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux
> * openmediavault version:  7.0.5-1
> 
> **Client (please complete the following information):**
> 
> * Device: PC
> * OS: Win 11
> * Browser firefox
> * Version 124

Have the same error just now, I can't access it since I don't have mini hdmi cable 😞

**Comment by Johboh** [comment]
Got the same issue now on 7.4.11-1 (Debian 12).
After changing hostname in the web UI (Network -> General), the machine was no longer accessible (not via web UI or SSH).
After connecting a screen, I could see that the ethernet card was no longer setup for DCHP, and thus not receiving any IP address. After re-enabling this I was able to access the machine again, but on reboot, I would loose DHCP.

I have two Ethernet ports on my machine (but only one connected), and when I installed OMV initially, I could see both listed under OMV -> Network -> Interfaces. Now after hostname change, one was missing (the connected one). After re-adding it in OMV and enabling DHCP, it does work again even after reboot.

So I don't know what happen here, if changing hostname for some reason removed/cleared/reset the configured network interfaces.

**Comment by fredtep** [comment]
Same here.
Just for the record here is a fix.
On a raspberry pi you can just plug hdmi and running omv-firstaid will help you rebuild the network.
On a rockpi it was a bit more tricky. Hdmi won't work so I had to use the console. For that you need a USB to TTL usb cable. This site will help : https://wiki.radxa.com/Rockpi4/dev/serial-console

**Comment by votdev** [comment]
INFO: Using the domain name [local](https://en.wikipedia.org/wiki/.local) is denied because this is reserved for mDNS. Use [internal](https://en.wikipedia.org/wiki/.internal) instead.

**Comment by LinuxfarmerHH** [comment]
OMV 7.7.3-1 (Sandworm)

Did not know about this issue, set a domainname and bada boooom.....

Headless pi was gone from the net, IPv4 not known anymore, same with last IPv6. No new dhcp request in my router, so cold boot. No change, pi did not ask for IP at the router. So it's headless and i have 3 drawers of computer cables, but none micro-hdmi.
Pulled the usb key, the OS is running on to investigate what happens on my PC.

Found in /etc/hosts
127.0.0.1 localhost
127.0.1.1 pi-name

as it's latest own ip adress set by OMV?

Searched through latest syslog, because it timestamp tell that the machine was running altough. There i found for the end0 network interface a 172. IPv4 added to the interface, a none routing that ends nowhere.

Conclusion: Without micro-hdmi only 'format c:' and reinstall everything from scratch solved this. I'm angry about such major showstoppers, how can an installed program destroy the linux host system?

Hope this bug will be adressed in some way. Buyin a micro-hdmi cable for the next round.

**Comment by votdev** [comment]
> OMV 7.7.3-1 (Sandworm)
> 
> Did not know about this issue, set a domainname and bada boooom.....
> 
> Headless pi was gone from the net, IPv4 not known anymore, same with last IPv6. No new dhcp request in my router, so cold boot. No change, pi did not ask for IP at the router. So it's headless and i have 3 drawers of computer cables, but none micro-hdmi. Pulled the usb key, the OS is running on to investigate what happens on my PC.
> 
> Found in /etc/hosts 127.0.0.1 localhost 127.0.1.1 pi-name
> 
> as it's latest own ip adress set by OMV?
> 
> Searched through latest syslog, because it timestamp tell that the machine was running altough. There i found for the end0 network interface a 172. IPv4 added to the interface, a none routing that ends nowhere.
> 
> Conclusion: Without micro-hdmi only 'format c:' and reinstall everything from scratch solved this. I'm angry about such major showstoppers, how can an installed program destroy the linux host system?
> 
> Hope this bug will be adressed in some way. Buyin a micro-hdmi cable for the next round.


The hosts file you are mentioning is not generated by OMV. A file that is generated by OMV looks like this https://github.com/openmediavault/openmediavault/blob/master/deb/openmediavault/srv/salt/omv/deploy/hosts/10default.sls.
So something else on a PI overwrite the file managed by OMV.
I do not have a PI, but I assume that a Debian derivate behaves like the origin. 
As OMV is an open source project, I expect the community to make its contribution and not just grumble and make demands.

**Comment by ryecoaaron** [comment]
Your /etc/hosts file didn't break nginx or ssh.  So, this did not cause the need to get cables out.

**Comment by LinuxfarmerHH** [comment]
With new raspbian install, i got the idea to setup a static IP first, before omv installation. Raspbian is changed in strange ways from their devs.
There is only dhclient installed, no network-manager. The option to change static / dhcp was removed from raspi-config. To get a static ip, the network manager has to be installed, to get hand on the nmtui tool. The network interface end0 is unused, they bridged something from lo to lo1 to jump around it. There is no internet connection in network manager, for static we need to generate the first connection. Never seen this with debian, ubuntu, fedora and arch linux before.

I don't know if such construction in raspbian is even compatible with omv.
